azurelinuxagent/common/protocol/wire.py

```diff
@@ -778,18 +776,30 @@ def update_host_plugin(self, container_id, role_config_name):
             self._host_plugin.update_container_id(container_id)
             self._host_plugin.update_role_config_name(role_config_name)
 
-    def update_goal_state(self, force_update=False, silent=False):
+    def update_goal_state(self, silent=False):
```

Comment:
I don't quite get the update_goal_state vs reset_goal_state. When do we use each of these?

Reply:
We're removing the force_update parameter and using reset_goal_state in all cases where force_update was True. 

Reset will initialize the goal state instead of updating